File: serialEventHdlr/scanner/doc/lidar_mock.py
import numpy as np
import math
import matplotlib.pyplot as plt

#from mpl_toolkits.mplot3d import axes3d
import matplotlib.pyplot as plt
from matplotlib import style

ANGLE_INC = 360/1600

alpha = np.arange(0, 360, ANGLE_INC)
BETA = 10

r = []
for a in alpha:
    try:
        r1 = 5/(math.cos(np.rad2deg(a))*math.cos(np.rad2deg(BETA)))
    except ZeroDivisionError:
        r1 = 1000 #something large

    try:
        r2 = 4/(math.sin(np.rad2deg(a))*math.cos(np.rad2deg(BETA)))
    except ZeroDivisionError:
        r2 = 1000 #something large

    r_temp = min(abs(r1), abs(r2))
    r.append(r_temp)

    #str = "r = {:.3f}, alpha = {:.3f}, beta = {:.2f}".format(r_temp, a, BETA)
    #print(str)

File: serialEventHdlr/scanner/doc/lidar_mock.m
% r is what will be read from lidar
angle_inc = 90/400;
alpha = 0:angle_inc:360-angle_inc; % angle in the horizontal plane
beta = 10;
r = min(abs( 5./(cosd(alpha)*cosd(beta)) ), abs( 4./(sind(alpha)*cosd(beta)) ));

x = r.*cosd(alpha)*cosd(beta);
y = r.*sind(alpha)*cosd(beta);
z = r*sind(beta);
xy = [x' y' z'];
%add z
%plannar_coord = [xy zeros(size(xy,1),1)];

plot3(xy(:,1), xy(:,2), xy(:,3), '*')
hold on
for i = 1:length(r)
    plot3([0 xy(i, 1)], [0 xy(i, 2)], [0 xy(i, 3)])
end

% alpha - vinkeln i horisontalplanet
% beta - vinkeln i vertikalplanet
%
% x = r*cos(alpha)*cos(beta)
% y = r*sin(alpha)*cos(beta)
% z = r*sin(beta)

%enligt det henrik länkade
%x = r * sin(th) * cos(tv)
%y = r * sin(th) * sin(tv)
%z = r * cos(th)